File: CatSero-Bukkit/src/main/java/moe/xmcn/catsero/v2/listeners/DispatchCommand/OnMiraiGroupMessageEvent.java
package moe.xmcn.catsero.v2.listeners.DispatchCommand;

import me.dreamvoid.miraimc.bukkit.event.message.passive.MiraiGroupMessageEvent;
import moe.xmcn.catsero.v2.utils.Configs;
import moe.xmcn.catsero.v2.utils.Env;
import moe.xmcn.catsero.v2.utils.Loggers;
import moe.xmcn.catsero.v2.utils.QPS;
import org.bukkit.Bukkit;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;

import java.util.UUID;

public class OnMiraiGroupMessageEvent implements Listener {

    @EventHandler
    public void onMiraiGroupMessageEvent(MiraiGroupMessageEvent e) {
        try {
            String[] strings = QPS.getParser.parse(e.getMessage());
            if (
                    Configs.JPConfig.uses_config.getBoolean("dispatch-command.enabled") &&
                            e.getBotID() == Configs.getBotCode(Utils.X_Bot) &&
                            e.getGroupID() == Configs.getGroupCode(Utils.X_Group) &&
                            strings != null &&
                            strings[0].equals("cmd")
            ) {
                if (Configs.isQQOp(e.getSenderID())) {
                    UUID uuid = UUID.randomUUID();
                    Bukkit.getScheduler().runTask(
                            Configs.plugin,
                            () -> {
                                Loggers.CustomLevel.logSign("start", uuid);
                                Bukkit.dispatchCommand(
                                        Bukkit.getConsoleSender(),
                                        Utils.iterateArray(strings)
                                );
                                Loggers.CustomLevel.logSign("end", uuid);
                            }
                    );
                    String data = Loggers.CustomLevel.getSignBlock(uuid);
                    System.out.println(data);
                    wait(2000);
                    Env.AMiraiMC.sendMiraiGroupMemberMessage(data, Utils.X_Bot, Utils.X_Group, e.getSenderID());
                    Env.AMiraiMC.sendMiraiGroupMessage(Configs.getMsgByMsID("qq.dispatch-command.success"), Utils.X_Bot, Utils.X_Group);
                } else
                    Env.AMiraiMC.sendMiraiGroupMessage(Configs.getMsgByMsID("qq.no-permission"), Utils.X_Bot, Utils.X_Group);
            }
        } catch (Exception ex) {
            Loggers.CustomLevel.logCatch(ex);
        }
    }

}

File: src/main/java/moe/xmcn/catsero/event/listener/AutoAnswer/onMiraiGroupMessageEvent.kt
package moe.xmcn.catsero.event.listener.AutoAnswer

import me.dreamvoid.miraimc.api.MiraiBot
import me.dreamvoid.miraimc.bukkit.event.MiraiGroupMessageEvent
import moe.xmcn.catsero.utils.Config
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener

class onMiraiGroupMessageEvent : Listener {
    @EventHandler
    fun onGroupMessage(event: MiraiGroupMessageEvent) {
        Config.Use_Bots.forEach { bot ->
            Config.Use_Groups.forEach { group ->
                if (Config.UsesConfig.getBoolean("auto-answer.enabled") && event.groupID == bot && event.botID == bot) {
                    val config = Config.customConfig(Config.UsesConfig.getString("auto-answer.data-file"))
                    config.getStringList("list").forEach { list ->
                        val ques = config.getString("$list.ques")
                        val answ = config.getString("$list.answ")
                        val msg = event.message
                        if (msg == ques) {
                            try {
                                MiraiBot.getBot(bot).getGroup(group).sendMessageMirai(answ)
                            } catch (nse: NoSuchElementException) {
                                println("发送消息时出现异常:\n$nse" + nse.stackTrace)
                            }
                        }
                    }
                }
            }
        }
    }
}